### ISSUE TITLE
Kernel Panics

### ISSUE THREAD

**Issue by shavenne** [multipath-tcp/mptcp#130]
Hi!

I'm not 100% sure if it's a MPTCP problem (but 95%).
I often have crashes. I tried another network card, changed motherboard, memory, cpu, changed to a virtualization, changed the distro (Ubuntu to Gentoo) but I still have them: Kernel panics. Sometimes after some minutes after boot, sometimes after a week.

> [ 8498.875359] BUG: unable to handle kernel paging request at 0000002000000000
> [ 8498.876506] IP: [<ffffffff811652d5>] kmem_cache_alloc_trace+0x65/0x150
> [ 8498.877527] PGD b04af067 PUD 0
> [ 8498.878088] Oops: 0000 [#1] SMP
> [ 8498.878661] Modules linked in: xt_REDIRECT nf_nat_redirect xt_nat ipt_MASQUERADE nf_nat_masquerade_ipv4 iptable_nat nf_nat_ipv4 nf_nat coretemp
> [ 8498.880888] CPU: 0 PID: 4825 Comm: snmpd Not tainted 4.1.26-hybrid #9
> [ 8498.881893] Hardware name: VMware, Inc. VMware Virtual Platform/440BX Desktop Reference Platform, BIOS 6.00 09/21/2015
> [ 8498.883579] task: ffff8800ba3d8000 ti: ffff8800b054c000 task.ti: ffff8800b054c000
> [ 8498.884783] RIP: 0010:[<ffffffff811652d5>]  [<ffffffff811652d5>] kmem_cache_alloc_trace+0x65/0x150
> [ 8498.886229] RSP: 0018:ffff8800b054fdb8  EFLAGS: 00010282
> [ 8498.887070] RAX: 0000000000000000 RBX: 0000000000000000 RCX: 0000000000457972
> [ 8498.888183] RDX: 0000000000457971 RSI: 0000000000008020 RDI: ffff8800bf803b00
> [ 8498.889288] RBP: ffff8800b054fdf8 R08: 0000000000019380 R09: ffff8800b721d01c
> [ 8498.890390] R10: 0000000000000000 R11: 0000000000000246 R12: 0000000000008020
> [ 8498.891520] R13: ffff8800bf803b00 R14: 0000002000000000 R15: ffffffff813ca896
> [ 8498.892628] FS:  00007ffa03a22700(0000) GS:ffff8800bb600000(0000) knlGS:0000000000000000
> [ 8498.893940] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
> [ 8498.894857] CR2: 0000002000000000 CR3: 00000000b5bca000 CR4: 00000000002406f0
> [ 8498.895981] Stack:
> [ 8498.896342]  00000000000080d0 0000000000000030 ffff8800b054fe08 0000000000000000
> [ 8498.897631]  ffff8800ae5c2e80 ffff8800ae5c2e80 00000000ffffff9f 0000000000000002
> [ 8498.898928]  ffff8800b054fe28 ffffffff813ca896 ffff88009b584100 ffff8800ba41ad00
> [ 8498.900211] Call Trace:
> [ 8498.900640]  [<ffffffff813ca896>] selinux_netlbl_sock_genattr+0x46/0xf0
> [ 8498.901672]  [<ffffffff813cad85>] selinux_netlbl_socket_post_create+0x35/0x80
> [ 8498.902806]  [<ffffffff813b717d>] selinux_socket_post_create+0x8d/0x1c0
> [ 8498.903851]  [<ffffffff813aed01>] security_socket_post_create+0x11/0x20
> [ 8498.904886]  [<ffffffff818552a6>] __sock_create+0x166/0x260
> [ 8498.905766]  [<ffffffff81856aed>] SyS_socket+0x4d/0xd0
> [ 8498.906587]  [<ffffffff81a6b6d7>] system_call_fastpath+0x12/0x6a
> [ 8498.907527] Code: 8b 45 00 65 49 8b 50 08 65 4c 03 05 76 4e ea 7e 4d 8b 30 4d 85 f6 74 5e 49 83 78 10 00 74 57 49 63 45 20 48 8d 4a 01 4d 8b 45 00 <49> 8b 1c 06 4c 89 f0 65 49 0f c7 08 0f 94 c0 84 c0 74 c1 49 63
> [ 8498.911929] RIP  [<ffffffff811652d5>] kmem_cache_alloc_trace+0x65/0x150
> [ 8498.912987]  RSP <ffff8800b054fdb8>
> [ 8498.913562] CR2: 0000002000000000
> [ 8498.914254] ---[ end trace fb5ec96c810e986f ]---
> [ 8498.915434] BUG: unable to handle kernel paging request at 0000002000000000
> [ 8498.916628] IP: [<ffffffff81164e9f>] kmem_cache_alloc+0x5f/0x160
> [ 8498.917666] PGD b8d10067 PUD 0
> [ 8498.918278] Oops: 0000 [#2] SMP
> [ 8498.918902] Modules linked in: xt_REDIRECT nf_nat_redirect xt_nat ipt_MASQUERADE nf_nat_masquerade_ipv4 iptable_nat nf_nat_ipv4 nf_nat coretemp
> [ 8498.921273] CPU: 0 PID: 24755 Comm: kworker/u128:0 Tainted: G      D         4.1.26-hybrid #9
> [ 8498.922649] Hardware name: VMware, Inc. VMware Virtual Platform/440BX Desktop Reference Platform, BIOS 6.00 09/21/2015
> [ 8498.924371] Workqueue: khelper __call_usermodehelper
> [ 8498.925220] task: ffff8800ba4c0910 ti: ffff8800a329c000 task.ti: ffff8800a329c000
> [ 8498.926439] RIP: 0010:[<ffffffff81164e9f>]  [<ffffffff81164e9f>] kmem_cache_alloc+0x5f/0x160
> [ 8498.927835] RSP: 0018:ffff8800a329fc18  EFLAGS: 00010282
> [ 8498.928688] RAX: 0000000000000000 RBX: ffff8800ba4c3660 RCX: 0000000000457974
> [ 8498.929808] RDX: 0000000000457973 RSI: 00000000000080d0 RDI: ffff8800bf803b00
> [ 8498.930933] RBP: ffff8800a329fc58 R08: 0000000000019380 R09: 0000000000000000
> [ 8498.932084] R10: 0000000000000000 R11: 0000000000000000 R12: 00000000000080d0
> [ 8498.933210] R13: ffffffff810e751d R14: ffff8800bf803b00 R15: 0000002000000000
> [ 8498.934384] FS:  0000000000000000(0000) GS:ffff8800bb600000(0000) knlGS:0000000000000000
> [ 8498.935710] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
> [ 8498.936667] CR2: 0000002000000000 CR3: 00000000b8f49000 CR4: 00000000002406f0
> [ 8498.937806] Stack:
> [ 8498.938182]  ffff8800a329fc28 ffffffff813ae761 ffff8800a329fc48 ffff8800ba4c3660
> [ 8498.939527]  0000000000000000 0000000000800111 ffff8800ba4c3660 ffff8800ba4c0910
> [ 8498.940876]  ffff8800a329fc78 ffffffff810e751d 0000000000000000 0000000000000000
> [ 8498.942234] Call Trace:
> [ 8498.942678]  [<ffffffff813ae761>] ? security_prepare_creds+0x11/0x20
> [ 8498.945551]  [<ffffffff810e751d>] __delayacct_tsk_init+0x1d/0x40
> [ 8498.946605]  [<ffffffff810510e8>] copy_process.part.43+0x5e8/0x18a0
> [ 8498.947604]  [<ffffffff810aaf83>] ? add_timer+0x13/0x20
> [ 8498.948447]  [<ffffffff810696a3>] ? __queue_delayed_work+0x83/0x1b0
> [ 8498.949472]  [<ffffffff8107f86d>] ? update_curr+0xad/0x160
> [ 8498.950359]  [<ffffffff81065df0>] ? __call_usermodehelper+0x70/0x70
> [ 8498.951353]  [<ffffffff8108403c>] ? dequeue_task_fair+0xbac/0xd00
> [ 8498.952329]  [<ffffffff81052537>] do_fork+0xb7/0x380
> [ 8498.953135]  [<ffffffff81080ae7>] ? put_prev_task_fair+0x27/0x40
> [ 8498.954107]  [<ffffffff81052821>] kernel_thread+0x21/0x30
> [ 8498.954988]  [<ffffffff81065dcc>] __call_usermodehelper+0x4c/0x70
> [ 8498.955962]  [<ffffffff81069e61>] process_one_work+0x151/0x430
> [ 8498.956899]  [<ffffffff8106a46b>] worker_thread+0x4b/0x470
> [ 8498.957783]  [<ffffffff8106a420>] ? rescuer_thread+0x2e0/0x2e0
> [ 8498.958720]  [<ffffffff8106f164>] kthread+0xc4/0xe0
> [ 8498.959513]  [<ffffffff8106f0a0>] ? kthread_create_on_node+0x180/0x180
> [ 8498.960552]  [<ffffffff81a6ba92>] ret_from_fork+0x42/0x70
> [ 8498.961425]  [<ffffffff8106f0a0>] ? kthread_create_on_node+0x180/0x180
> [ 8498.962475] Code: 00 4d 8b 06 65 49 8b 50 08 65 4c 03 05 ab 52 ea 7e 4d 8b 38 4d 85 ff 74 6b 49 83 78 10 00 74 64 49 63 46 20 48 8d 4a 01 4d 8b 06 <49> 8b 1c 07 4c 89 f8 65 49 0f c7 08 0f 94 c0 84 c0 74 c3 49 63
> [ 8498.967849] RIP  [<ffffffff81164e9f>] kmem_cache_alloc+0x5f/0x160
> [ 8498.968861]  RSP <ffff8800a329fc18>
> [ 8498.969451] CR2: 0000002000000000
> [ 8498.970017] ---[ end trace fb5ec96c810e9870 ]---
> [ 8498.971000] BUG: unable to handle kernel [ 8498.971002] IP: [<ffffffff81164e9f>] kmem_cache_alloc+0x5f/0x160
> [ 8498.971005] PGD b06cf067 PUD 0
> [ 8498.971006] Oops: 0000 [#3] SMP
> [ 8498.971007] Modules linked in: xt_REDIRECT nf_nat_redirect xt_nat ipt_MASQUERADE nf_nat_masquerade_ipv4 iptable_nat nf_nat_ipv4 nf_nat coretemp
> [ 8498.971010] CPU: 0 PID: 5014 Comm: redsocks.sh Tainted: G      D         4.1.26-hybrid #9
> [ 8498.971011] Hardware name: VMware, Inc. VMware Virtual Platform/440BX Desktop Reference Platform, BIOS 6.00 09/21/2015
> [ 8498.971011] task: ffff8800ba7a8000 ti: ffff8800b70f8000 task.ti: ffff8800b70f8000
> [ 8498.971012] RIP: 0010:[<ffffffff81164e9f>]  [<ffffffff81164e9f>] kmem_cache_alloc+0x5f/0x160
> [ 8498.971014] RSP: 0018:ffff8800b70fbd68  EFLAGS: 00010282
> [ 8498.971014] RAX: 0000000000000000 RBX: ffff8800a3047590 RCX: 000000000045798c
> [ 8498.971014] RDX: 000000000045798b RSI: 00000000000000d0 RDI: ffff8800bf803b00
> [ 8498.971015] RBP: ffff8800b70fbda8 R08: 0000000000019380 R09: 0000000000000b67
> [ 8498.971015] R10: 0000000000000ae4 R11: ffff8800aaf49880 R12: 00000000000000d0
> [ 8498.971015] R13: ffffffff8114e553 R14: ffff8800bf803b00 R15: 0000002000000000
> [ 8498.971016] FS:  00007f7052fc6700(0000) GS:ffff8800bb600000(0000) knlGS:0000000000000000
> [ 8498.971017] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
> [ 8498.971037] CR2: 0000002000000000 CR3: 00000000b07b3000 CR4: 00000000002406f0
> [ 8498.971055] Stack:
> [ 8498.971056]  ffff8800a3047608 ffffffff8114e521 0000000000000001 ffff8800a3047590
> [ 8498.971057]  ffff8800b5ad1700 0000000000000000 ffff8800a31ba640 ffff8800b5ad1700
> [ 8498.971058]  ffff8800b70fbde8 ffffffff8114e553 ffff8800b70fbdd8 ffff8800ae722680
> [ 8498.971058] Call Trace:
> [ 8498.971061]  [<ffffffff8114e521>] ? anon_vma_fork+0x71/0x140
> [ 8498.971062]  [<ffffffff8114e553>] anon_vma_fork+0xa3/0x140
> [ 8498.971064]  [<ffffffff81051535>] copy_process.part.43+0xa35/0x18a0
> [ 8498.971066]  [<ffffffff8105de3a>] ? recalc_sigpending+0x1a/0x50
> [ 8498.971067]  [<ffffffff81052537>] do_fork+0xb7/0x380
> [ 8498.971068]  [<ffffffff8106116f>] ? sigprocmask+0x4f/0x80
> [ 8498.971069]  [<ffffffff81052881>] SyS_clone+0x11/0x20
> [ 8498.971071]  [<ffffffff81a6b6d7>] system_call_fastpath+0x12/0x6a
> [ 8498.971071] Code: 00 4d 8b 06 65 49 8b 50 08 65 4c 03 05 ab 52 ea 7e 4d 8b 38 4d 85 ff 74 6b 49 83 78 10 00 74 64 49 63 46 20 48 8d 4a 01 4d 8b 06 <49> 8b 1c 07 4c 89 f8 65 49 0f c7 08 0f 94 c0 84 c0 74 c3 49 63
> [ 8498.971081] RIP  [<ffffffff81164e9f>] kmem_cache_alloc+0x5f/0x160
> [ 8498.971082]  RSP <ffff8800b70fbd68>
> [ 8498.971083] CR2: 0000002000000000
> [ 8498.971083] ---[ end trace fb5ec96c810e9871 ]---
> [ 8498.971215] BUG: unable to handle kernel paging request at ffffffffffffffd8
> [ 8498.971216] IP: [<ffffffff8106f77b>] kthread_data+0xb/0x20
> [ 8498.971218] PGD 200c067 PUD 200e067 PMD 0
> [ 8498.971220] Oops: 0000 [#4] SMP
> [ 8498.971220] Modules linked in: xt_REDIRECT nf_nat_redirect xt_nat ipt_MASQUERADE nf_nat_masquerade_ipv4 iptable_nat nf_nat_ipv4 nf_nat coretemp
> [ 8498.971223] CPU: 0 PID: 24755 Comm: kworker/u128:0 Tainted: G      D         4.1.26-hybrid #9
> [ 8498.971224] Hardware name: VMware, Inc. VMware Virtual Platform/440BX Desktop Reference Platform, BIOS 6.00 09/21/2015
> [ 8498.971229] task: ffff8800ba4c0910 ti: ffff8800a329c000 task.ti: ffff8800a329c000
> [ 8498.971230] RIP: 0010:[<ffffffff8106f77b>]  [<ffffffff8106f77b>] kthread_data+0xb/0x20
> [ 8498.971232] RSP: 0018:ffff8800a329f8c8  EFLAGS: 00010096
> [ 8498.971232] RAX: 0000000000000000 RBX: 0000000000000000 RCX: 000000000000000d
> [ 8498.971232] RDX: 000000000000000d RSI: 0000000000000000 RDI: ffff8800ba4c0910
> [ 8498.971233] RBP: ffff8800a329f8c8 R08: 0000000000000000 R09: 000000000000ba73
> [ 8498.971233] R10: 0000000000000260 R11: 0000000000000000 R12: 0000000000016200
> [ 8498.971233] R13: ffff8800bb616200 R14: ffff8800ba4c0910 R15: 0000000000000000
> [ 8498.971234] FS:  0000000000000000(0000) GS:ffff8800bb600000(0000) knlGS:0000000000000000
> [ 8498.971234] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
> [ 8498.971236] CR2: 0000000000000028 CR3: 00000000ae60d000 CR4: 00000000002406f0
> [ 8498.971236] Stack:
> [ 8498.971237]  ffff8800a329f8e8 ffffffff8106a900 ffff8800a329f8e8 ffff8800bb616200
> [ 8498.971237]  ffff8800a329f948 ffffffff81a67abd ffff880000000000 ffff8800ba4c0910
> [ 8498.971238]  ffff8800aa8dba80 ffff8800aa8dba90 ffff8800a329f948 ffff8800a32a0000
> [ 8498.971239] Call Trace:
> [ 8498.971240]  [<ffffffff8106a900>] wq_worker_sleeping+0x10/0xa0
> [ 8498.971242]  [<ffffffff81a67abd>] __schedule+0x83d/0xa50
> [ 8498.971242]  [<ffffffff81a67d02>] schedule+0x32/0x80
> [ 8498.971244]  [<ffffffff81054af6>] do_exit+0x766/0xb70
> [ 8498.971245]  [<ffffffff81006eb8>] oops_end+0x88/0xd0
> [ 8498.971247]  [<ffffffff81047079>] no_context+0x109/0x360
> [ 8498.971248]  [<ffffffff81136038>] ? zone_statistics+0x78/0x90
> [ 8498.971249]  [<ffffffff81047345>] __bad_area_nosemaphore+0x75/0x200
> [ 8498.971250]  [<ffffffff810474de>] bad_area_nosemaphore+0xe/0x10
> [ 8498.971251]  [<ffffffff8104777e>] __do_page_fault+0xae/0x440
> [ 8498.971253]  [<ffffffff810e751d>] ? __delayacct_tsk_init+0x1d/0x40
> [ 8498.971254]  [<ffffffff81047b1c>] do_page_fault+0xc/0x10
> [ 8498.971256]  [<ffffffff81a6d0f2>] page_fault+0x22/0x30
> [ 8498.971257]  [<ffffffff810e751d>] ? __delayacct_tsk_init+0x1d/0x40
> [ 8498.971258]  [<ffffffff81164e9f>] ? kmem_cache_alloc+0x5f/0x160
> [ 8498.971259]  [<ffffffff81164e6c>] ? kmem_cache_alloc+0x2c/0x160
> [ 8498.971261]  [<ffffffff813ae761>] ? security_prepare_creds+0x11/0x20
> [ 8498.971262]  [<ffffffff810e751d>] __delayacct_tsk_init+0x1d/0x40
> [ 8498.971263]  [<ffffffff810510e8>] copy_process.part.43+0x5e8/0x18a0
> [ 8498.971265]  [<ffffffff810aaf83>] ? add_timer+0x13/0x20
> [ 8498.971266]  [<ffffffff810696a3>] ? __queue_delayed_work+0x83/0x1b0
> [ 8498.971268]  [<ffffffff8107f86d>] ? update_curr+0xad/0x160
> [ 8498.971269]  [<ffffffff81065df0>] ? __call_usermodehelper+0x70/0x70
> [ 8498.971270]  [<ffffffff8108403c>] ? dequeue_task_fair+0xbac/0xd00
> [ 8498.971271]  [<ffffffff81052537>] do_fork+0xb7/0x380
> [ 8498.971272]  [<ffffffff81080ae7>] ? put_prev_task_fair+0x27/0x40
> [ 8498.971273]  [<ffffffff81052821>] kernel_thread+0x21/0x30
> [ 8498.971274]  [<ffffffff81065dcc>] __call_usermodehelper+0x4c/0x70
> [ 8498.971275]  [<ffffffff81069e61>] process_one_work+0x151/0x430
> [ 8498.971276]  [<ffffffff8106a46b>] worker_thread+0x4b/0x470
> [ 8498.971277]  [<ffffffff8106a420>] ? rescuer_thread+0x2e0/0x2e0
> [ 8498.971278]  [<ffffffff8106f164>] kthread+0xc4/0xe0
> [ 8498.971280]  [<ffffffff8106f0a0>] ? kthread_create_on_node+0x180/0x180
> [ 8498.971281]  [<ffffffff81a6ba92>] ret_from_fork+0x42/0x70
> [ 8498.971282]  [<ffffffff8106f0a0>] ? kthread_create_on_node+0x180/0x180
> [ 8498.971282] Code: 00 48 89 e5 5d 48 8b 40 c8 48 c1 e8 02 83 e0 01 c3 0f 1f 44 00 00 66 2e 0f 1f 84 00 00 00 00 00 48 8b 87 e0 04 00 00 55 48 89 e5 <48> 8b 40 d8 5d c3 0f 1f 44 00 00 66 2e 0f 1f 84 00 00 00 00 00
> [ 8498.971292] RIP  [<ffffffff8106f77b>] kthread_data+0xb/0x20
> [ 8498.971294]  RSP <ffff8800a329f8c8>
> [ 8498.971294] CR2: ffffffffffffffd8
> [ 8498.971294] ---[ end trace fb5ec96c810e9872 ]---
> [ 8498.971295] Fixing recursive fault but reboot is needed!
> paging request at 0000002000000000
> [ 8519.999990] INFO: rcu_sched detected stalls on CPUs/tasks:
> [ 8520.010580]  0: (1 GPs behind) idle=ecb/140000000000000/0 softirq=468584/468585 fqs=6979
> [ 8520.012606](detected by 1, t=21002 jiffies, g=183534, c=183533, q=1032)
> [ 8520.013708] Task dump for CPU 0:
> [ 8520.014255] kworker/u128:0  D 0000000000010096 12160 24755      0 0x00000008
> [ 8520.015430]  ffff8800a329f8c8 0000000000000018 0000002000000000 ffff8800a329f8e8
> [ 8520.016761]  ffffffff8106a900 ffff8800a329f8e8 ffff8800bb616200 ffff8800a329f948
> [ 8520.018074]  ffffffff81a67abd ffff880000000000 ffff8800ba4c0910 ffff8800aa8dba80
> [ 8520.019389] Call Trace:
> [ 8520.019826]  [<ffffffff8106a900>] wq_worker_sleeping+0x10/0xa0
> [ 8520.020741]  [<ffffffff81a67abd>] __schedule+0x83d/0xa50
> [ 8520.021575]  [<ffffffff81a67d02>] schedule+0x32/0x80
> [ 8520.022359]  [<ffffffff81054af6>] do_exit+0x766/0xb70
> [ 8520.023894]  [<ffffffff81006eb8>] oops_end+0x88/0xd0
> [ 8520.024687]  [<ffffffff81047079>] no_context+0x109/0x360
> [ 8520.025521]  [<ffffffff81136038>] ? zone_statistics+0x78/0x90
> [ 8520.026425]  [<ffffffff81047345>] __bad_area_nosemaphore+0x75/0x200
> [ 8520.027434]  [<ffffffff810474de>] bad_area_nosemaphore+0xe/0x10
> [ 8520.028357]  [<ffffffff8104777e>] __do_page_fault+0xae/0x440
> [ 8520.029246]  [<ffffffff810e751d>] ? __delayacct_tsk_init+0x1d/0x40
> [ 8520.030202]  [<ffffffff81047b1c>] do_page_fault+0xc/0x10
> [ 8520.031034]  [<ffffffff81a6d0f2>] page_fault+0x22/0x30
> [ 8520.031846]  [<ffffffff810e751d>] ? __delayacct_tsk_init+0x1d/0x40
> [ 8520.032803]  [<ffffffff81164e9f>] ? kmem_cache_alloc+0x5f/0x160
> [ 8520.033721]  [<ffffffff81164e6c>] ? kmem_cache_alloc+0x2c/0x160
> [ 8520.034643]  [<ffffffff813ae761>] ? security_prepare_creds+0x11/0x20
> [ 8520.035626]  [<ffffffff810e751d>] __delayacct_tsk_init+0x1d/0x40
> [ 8520.037196]  [<ffffffff810510e8>] copy_process.part.43+0x5e8/0x18a0
> [ 8520.038207]  [<ffffffff810aaf83>] ? add_timer+0x13/0x20
> [ 8520.039030]  [<ffffffff810696a3>] ? __queue_delayed_work+0x83/0x1b0
> [ 8520.040004]  [<ffffffff8107f86d>] ? update_curr+0xad/0x160
> [ 8520.040862]  [<ffffffff81065df0>] ? __call_usermodehelper+0x70/0x70
> [ 8520.041831]  [<ffffffff8108403c>] ? dequeue_task_fair+0xbac/0xd00
> [ 8520.042779]  [<ffffffff81052537>] do_fork+0xb7/0x380
> [ 8520.043564]  [<ffffffff81080ae7>] ? put_prev_task_fair+0x27/0x40
> [ 8520.044494]  [<ffffffff81052821>] kernel_thread+0x21/0x30
> [ 8520.045340]  [<ffffffff81065dcc>] __call_usermodehelper+0x4c/0x70
> [ 8520.046288]  [<ffffffff81069e61>] process_one_work+0x151/0x430
> [ 8520.047197]  [<ffffffff8106a46b>] worker_thread+0x4b/0x470
> [ 8520.048056]  [<ffffffff8106a420>] ? rescuer_thread+0x2e0/0x2e0
> [ 8520.048967]  [<ffffffff8106f164>] kthread+0xc4/0xe0
> [ 8520.049739]  [<ffffffff8106f0a0>] ? kthread_create_on_node+0x180/0x180
> [ 8520.051401]  [<ffffffff81a6ba92>] ret_from_fork+0x42/0x70
> [ 8520.052253]  [<ffffffff8106f0a0>] ? kthread_create_on_node+0x180/0x180

I have them since I'm using MPTCP and I'm absolutely clueless at the moment.
I'm using the hybrid-sources, Linux 4.1.26-hybrid. Should be version 0.91? I'm not sure.

Any additional information you need from me?
Maybe my iptables settings are relavant?

> atom ~ # iptables -t nat -S
> -P PREROUTING ACCEPT
> -P INPUT ACCEPT
> -P OUTPUT ACCEPT
> -P POSTROUTING ACCEPT
> -N FORWARDINGS
> -N REDSOCKS
> -A PREROUTING -i ppp0 -j FORWARDINGS
> -A PREROUTING -i ppp1 -j FORWARDINGS
> -A PREROUTING -i enp4s0f0 -j FORWARDINGS
> -A PREROUTING -i enp4s0f1 -j FORWARDINGS
> -A PREROUTING -i tap0 -j FORWARDINGS
> -A PREROUTING -i enp3s0f0 -p tcp -j REDSOCKS
> -A POSTROUTING -s 192.168.90.0/24 -o enp3s0f1 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o enp4s0f0 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o enp4s0f1 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o ppp0 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o ppp1 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o tap0 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o tun0 -j MASQUERADE
> -A POSTROUTING -s 192.168.99.0/24 -o ppp0 -j MASQUERADE
> -A POSTROUTING -s 192.168.99.0/24 -o ppp1 -j MASQUERADE
> -A FORWARDINGS -p tcp -m tcp --dport 8000 -j DNAT --to-destination 192.168.90.21:8000
> -A FORWARDINGS -p tcp -m tcp --dport 8443 -j DNAT --to-destination 192.168.90.101:443
> -A FORWARDINGS -p tcp -m tcp --dport 9443 -j DNAT --to-destination 192.168.90.102:443
> -A FORWARDINGS -p tcp -m tcp --dport 35555:35560 -j DNAT --to-destination 192.168.90.21
> -A FORWARDINGS -p udp -m udp --dport 3478 -j DNAT --to-destination 192.168.90.81
> -A FORWARDINGS -p udp -m udp --dport 4379:4380 -j DNAT --to-destination 192.168.90.81
> -A FORWARDINGS -p udp -m udp --dport 27000:27036 -j DNAT --to-destination 192.168.90.81
> -A FORWARDINGS -p udp -m udp --dport 27014:27050 -j DNAT --to-destination 192.168.90.81
> -A FORWARDINGS -p udp -m udp --dport 3074 -j DNAT --to-destination 192.168.90.81
> -A REDSOCKS -d 0.0.0.0/8 -j RETURN
> -A REDSOCKS -d 10.0.0.0/8 -j RETURN
> -A REDSOCKS -d 127.0.0.0/8 -j RETURN
> -A REDSOCKS -d 169.254.0.0/16 -j RETURN
> -A REDSOCKS -d 172.16.0.0/12 -j RETURN
> -A REDSOCKS -d 192.168.0.0/16 -j RETURN
> -A REDSOCKS -d 224.0.0.0/4 -j RETURN
> -A REDSOCKS -d 240.0.0.0/4 -j RETURN
> -A REDSOCKS -p tcp -m iprange --src-range 192.168.90.200-192.168.90.230 -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -p tcp -m iprange --src-range 192.168.90.200-192.168.90.230 -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.21/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.21/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.81/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.81/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.81/32 -p tcp -m tcp --dport 8888 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.81/32 -p tcp -m tcp --dport 8080 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.155/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.155/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.157/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.157/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.82/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.82/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.22/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.22/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.31/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.31/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.32/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.32/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> atom ~ # iptables -t mangle -S
> -P PREROUTING ACCEPT
> -P INPUT ACCEPT
> -P FORWARD ACCEPT
> -P OUTPUT ACCEPT
> -P POSTROUTING ACCEPT
> -A PREROUTING -m state --state RELATED,ESTABLISHED -j CONNMARK --restore-mark --nfmask 0xffffffff --ctmask 0xffffffff
> -A PREROUTING -i ppp0 -m state --state NEW -j CONNMARK --set-xmark 0xb/0xffffffff
> -A PREROUTING -i ppp1 -m state --state NEW -j CONNMARK --set-xmark 0xc/0xffffffff
> -A PREROUTING -i enp4s0f1 -m state --state NEW -j CONNMARK --set-xmark 0xd/0xffffffff
> -A PREROUTING -i tap0 -m state --state NEW -j CONNMARK --set-xmark 0xa/0xffffffff
> -A PREROUTING -m connmark --mark 0xa -j MARK --set-xmark 0xa/0xffffffff
> -A PREROUTING -m connmark --mark 0xb -j MARK --set-xmark 0xb/0xffffffff
> -A PREROUTING -m connmark --mark 0xc -j MARK --set-xmark 0xc/0xffffffff
> -A PREROUTING -m connmark --mark 0xd -j MARK --set-xmark 0xd/0xffffffff
> -A PREROUTING -m state --state NEW -m connmark ! --mark 0x0 -j CONNMARK --save-mark --nfmask 0xffffffff --ctmask 0xffffffff
> -A FORWARD -o ppp0 -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1400:65495 -j TCPMSS --clamp-mss-to-pmtu
> -A FORWARD -o ppp1 -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1400:65495 -j TCPMSS --clamp-mss-to-pmtu
> -A OUTPUT -m state --state RELATED,ESTABLISHED -j CONNMARK --restore-mark --nfmask 0xffffffff --ctmask 0xffffffff
> atom ~ # iptables -S
> -P INPUT ACCEPT
> -P FORWARD ACCEPT
> -P OUTPUT ACCEPT
> -A INPUT -d 224.0.0.0/4 -j ACCEPT
> -A FORWARD -d 224.0.0.0/4 -j ACCEPT
> -A FORWARD -s 93.186.0.0/16 -d 224.0.0.0/4 -j ACCEPT
> -A FORWARD -s 91.11.128.0/18 -d 224.0.0.0/4 -j ACCEPT
> -A FORWARD -s 193.158.35.0/24 -d 224.0.0.0/4 -j ACCEPT
> -A FORWARD -s 217.0.119.0/24 -d 224.0.0.0/4 -j ACCEPT
> atom ~ # iptables -t nat -S
> -P PREROUTING ACCEPT
> -P INPUT ACCEPT
> -P OUTPUT ACCEPT
> -P POSTROUTING ACCEPT
> -N FORWARDINGS
> -N REDSOCKS
> -A PREROUTING -i ppp0 -j FORWARDINGS
> -A PREROUTING -i ppp1 -j FORWARDINGS
> -A PREROUTING -i enp4s0f0 -j FORWARDINGS
> -A PREROUTING -i enp4s0f1 -j FORWARDINGS
> -A PREROUTING -i tap0 -j FORWARDINGS
> -A PREROUTING -i enp3s0f0 -p tcp -j REDSOCKS
> -A POSTROUTING -s 192.168.90.0/24 -o enp3s0f1 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o enp4s0f0 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o enp4s0f1 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o ppp0 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o ppp1 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o tap0 -j MASQUERADE
> -A POSTROUTING -s 192.168.90.0/24 -o tun0 -j MASQUERADE
> -A POSTROUTING -s 192.168.99.0/24 -o ppp0 -j MASQUERADE
> -A POSTROUTING -s 192.168.99.0/24 -o ppp1 -j MASQUERADE
> -A FORWARDINGS -p tcp -m tcp --dport 8000 -j DNAT --to-destination 192.168.90.21:8000
> -A FORWARDINGS -p tcp -m tcp --dport 8443 -j DNAT --to-destination 192.168.90.101:443
> -A FORWARDINGS -p tcp -m tcp --dport 9443 -j DNAT --to-destination 192.168.90.102:443
> -A FORWARDINGS -p tcp -m tcp --dport 35555:35560 -j DNAT --to-destination 192.168.90.21
> -A FORWARDINGS -p udp -m udp --dport 3478 -j DNAT --to-destination 192.168.90.81
> -A FORWARDINGS -p udp -m udp --dport 4379:4380 -j DNAT --to-destination 192.168.90.81
> -A FORWARDINGS -p udp -m udp --dport 27000:27036 -j DNAT --to-destination 192.168.90.81
> -A FORWARDINGS -p udp -m udp --dport 27014:27050 -j DNAT --to-destination 192.168.90.81
> -A FORWARDINGS -p udp -m udp --dport 3074 -j DNAT --to-destination 192.168.90.81
> -A REDSOCKS -d 0.0.0.0/8 -j RETURN
> -A REDSOCKS -d 10.0.0.0/8 -j RETURN
> -A REDSOCKS -d 127.0.0.0/8 -j RETURN
> -A REDSOCKS -d 169.254.0.0/16 -j RETURN
> -A REDSOCKS -d 172.16.0.0/12 -j RETURN
> -A REDSOCKS -d 192.168.0.0/16 -j RETURN
> -A REDSOCKS -d 224.0.0.0/4 -j RETURN
> -A REDSOCKS -d 240.0.0.0/4 -j RETURN
> -A REDSOCKS -p tcp -m iprange --src-range 192.168.90.200-192.168.90.230 -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -p tcp -m iprange --src-range 192.168.90.200-192.168.90.230 -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.21/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.21/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.81/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.81/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.81/32 -p tcp -m tcp --dport 8888 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.81/32 -p tcp -m tcp --dport 8080 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.155/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.155/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.157/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.157/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.82/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.82/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.22/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.22/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.31/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.31/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.32/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 12345
> -A REDSOCKS -s 192.168.90.32/32 -p tcp -m tcp --dport 443 -j REDIRECT --to-ports 12345
> atom ~ # iptables -t mangle -S
> -P PREROUTING ACCEPT
> -P INPUT ACCEPT
> -P FORWARD ACCEPT
> -P OUTPUT ACCEPT
> -P POSTROUTING ACCEPT
> -A PREROUTING -m state --state RELATED,ESTABLISHED -j CONNMARK --restore-mark --nfmask 0xffffffff --ctmask 0xffffffff
> -A PREROUTING -i ppp0 -m state --state NEW -j CONNMARK --set-xmark 0xb/0xffffffff
> -A PREROUTING -i ppp1 -m state --state NEW -j CONNMARK --set-xmark 0xc/0xffffffff
> -A PREROUTING -i enp4s0f1 -m state --state NEW -j CONNMARK --set-xmark 0xd/0xffffffff
> -A PREROUTING -i tap0 -m state --state NEW -j CONNMARK --set-xmark 0xa/0xffffffff
> -A PREROUTING -m connmark --mark 0xa -j MARK --set-xmark 0xa/0xffffffff
> -A PREROUTING -m connmark --mark 0xb -j MARK --set-xmark 0xb/0xffffffff
> -A PREROUTING -m connmark --mark 0xc -j MARK --set-xmark 0xc/0xffffffff
> -A PREROUTING -m connmark --mark 0xd -j MARK --set-xmark 0xd/0xffffffff
> -A PREROUTING -m state --state NEW -m connmark ! --mark 0x0 -j CONNMARK --save-mark --nfmask 0xffffffff --ctmask 0xffffffff
> -A FORWARD -o ppp0 -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1400:65495 -j TCPMSS --clamp-mss-to-pmtu
> -A FORWARD -o ppp1 -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1400:65495 -j TCPMSS --clamp-mss-to-pmtu
> -A OUTPUT -m state --state RELATED,ESTABLISHED -j CONNMARK --restore-mark --nfmask 0xffffffff --ctmask 0xffffffff
> atom ~ #

Hope you find something to fix this. :/

**Comment by cpaasch** [comment]
Do you have this crash every time you attempt an MPTCP connection, or does it happen only after a lot of traffic has passed?

**Comment by shavenne** [comment]
It seems absolutely random but it never happened on high traffic (yet).
Interesting is that it's often happens when exactly I come or leave home (because of bad wireless signal or something?). I don't know if it can be related. But the machine that crashes (== my router as you will have gathered) has no wireless device.
A SSH-connection over MPTCP is always open. My router connects via SSH to my rented server which also has MPTCP enabled. All my HTTP(S) traffic is going over this SSH connection to bond my internet connections (2x VDSL + LTE).

**Comment by cpaasch** [comment]
Do you have logs that say something like:

```
Attempt to release TCP socket in state
```

?

**Comment by shavenne** [comment]
Not to my knowledge.

**Comment by cpaasch** [comment]
Ok. It seems like the memory is corrupted. Are you able to get a coredump from the next time the kernel crashes? This would help to see why the memory got corrupted.

What are you running on your router? Is this openwrt?

**Comment by shavenne** [comment]
No, it's just a Gentoo Linux. Same problems with Ubuntu before.

I'll report when the next crash happens.

**Comment by cpaasch** [comment]
I will need a coredump, and the vmlinux compiled with debug_info enabled.

See https://wiki.gentoo.org/wiki/Kernel_Crash_Dumps for info on how to configure it.

Thanks!

**Comment by shavenne** [comment]
It's prepared. I'm waiting for the crash now.
In the morning today it happened again  two times while I was sleeping. Of course it was before I managed the coredump thing x_X. The first time it rebooted automatically. After 12min it crashed again, this time without reboot. :(

[crash_messages.txt](https://github.com/multipath-tcp/mptcp/files/355983/crash_messages.txt)

**Comment by shavenne** [comment]
Any news yet? Still having that problems :(. My router just had a crash again some minutes ago.

**Comment by Eibborest** [comment]
By any chance are you running debian 8 amd64?

**Comment by shavenne** [comment]
I also had Debian (don't know if it was 8) before I tried Ubuntu and Gentoo and had the same problems. At the moment the router has an uptime of 30 days without any crash. I didn't change anything. Seems to be veeery random.

Interesting fact: Because I use mptcp to bond my internet connections I have a rented rootserver with mptcp on the other side. Ubuntu 14.04, 3.18.20-90-mptcp, uptime: 212 days. It runs absolutely stable.

**Comment by cpaasch** [comment]
The above mentioned commit solves a memory-corruption issue. This might have been the reason for your issues here. Closing.

Please create a new issue (with coredump) if you see this happen again on an unmodified kernel.
